xchange-lgo/src/main/java/info/bitrich/xchangestream/lgo/LgoUserBatchSubscription.java

```diff
@@ -0,0 +1,102 @@
+package info.bitrich.xchangestream.lgo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.lgo.domain.*;
+import info.bitrich.xchangestream.lgo.dto.*;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import io.reactivex.Observable;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.*;
+import org.slf4j.*;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+
+class LgoUserBatchSubscription {
+
+    private final Observable<LgoGroupedUserUpdate> downstream;
+    private static final Logger LOGGER = LoggerFactory.getLogger(LgoUserBatchSubscription.class);
+    private final LgoStreamingService streamingService;
+    private final CurrencyPair currencyPair;
+
+
+    static LgoUserBatchSubscription create(LgoStreamingService streamingService, CurrencyPair currencyPair) {
+        return new LgoUserBatchSubscription(streamingService, currencyPair);
+    }
+
+    private LgoUserBatchSubscription(LgoStreamingService streamingService, CurrencyPair currencyPair) {
+        this.streamingService = streamingService;
+        this.currencyPair = currencyPair;
+        downstream = createSubscription();
+    }
+
+    Observable<LgoGroupedUserUpdate> getPublisher() {
+        return downstream;
+    }
+
+    private Observable<LgoGroupedUserUpdate> createSubscription() {
+        final ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+        return streamingService
+                .subscribeChannel(LgoAdapter.channelName("user", currencyPair))
+                .map(s -> mapper.readValue(s.toString(), LgoUserMessage.class))
+                .scan(new LgoGroupedUserUpdate(), (acc, s) -> {
+                    List<LgoBatchOrderEvent> events = new ArrayList<>();
+                    if (s.getType().equals("update")) {
+                        if (s.getBatchId() != acc.getBatchId() + 1) {
```

Comment:
Same thoughts around this one as the others.

Reply:
Which one exactly ? 

Reply:
I can't see the reason for resubscribing. You're clearly still getting messages. You might have missed one (which is fine, we don't care) or got one out of order (which is is easy to detect and ignore as discussed above).

Reply:
removed

Reply:
@badgerwithagun Just a note here, in offline conversations with @BodySplash, I found that some messages were being dropped.  I have added this back in my local version, no recommendation yet, since i have not re-reproduced.

Reply:
Leaving this conversation open pending another PR with fixes.